### DISCUSSION TITLE
Random sequence sampling from a time series for training fixed size input models

### DISCUSSION THREAD

**Discussion by DAT-FYAYC** [Nixtla/mlforecast#96] [General]
Does the framework provide random sequence sampling from a time series to use for training for fixed input size models?
Given we have a ts with 100 data points and I want to train a model with a fixed input size (e.g. TFT with encoder length of 6) and a fixed prediction horizon (h=3). When training the model it would make sense to take a random subsequence of size 9 (6+3) within the range of the available 100 data points (for each epoch) to ensure the model is getting trained over the full time period to capture all information within the time series. Is this already implemented when training such models?

**Comment by jmoralez** [comment]
This seems to be more in the scope of [neuralforecast](https://github.com/Nixtla/neuralforecast), have you taken a look there?

**Reply by DAT-FYAYC** [reply]
@jmoralez that´s right. I am sorry, wanted to post in in neuralforecast and will do so.
